dae/dae/genomic_resources/reference_genome.py

```diff
@@ -25,6 +29,210 @@
 logger = logging.getLogger(__name__)
 
 
+class GenomeStatisticsMixin:
+    @staticmethod
+    def get_global_statistic_file():
+        return "reference_genome_statistic.yaml"
+
+    @staticmethod
+    def get_chrom_file(chrom):
+        return f"{chrom}_statistic.yaml"
+
+
+class ReferenceGenomeStatistics(
+    ResourceStatistics,
+    GenomeStatisticsMixin
+):
+    """Class for accessing reference genome statistics."""
+
+    def __init__(self, resource_id, global_statistic, chrom_statistics):
+        super().__init__(resource_id)
+        self.global_statistic = global_statistic
+        self.chrom_statistics = chrom_statistics
+
+    @staticmethod
+    def build_statistics(genomic_resource):
+        chrom_statistics = {}
+        try:
+            global_stat_filepath = os.path.join(
+                ReferenceGenomeStatistics.get_statistics_folder(),
+                ReferenceGenomeStatistics.get_global_statistic_file()
+            )
+            with genomic_resource.open_raw_file(
+                    global_stat_filepath, mode="r") as infile:
+                global_statistic = GenomeStatistic.deserialize(infile.read())
+            for chrom in global_statistic.chromosomes:
+                chrom_stat_filepath = os.path.join(
+                    ReferenceGenomeStatistics.get_statistics_folder(),
+                    ReferenceGenomeStatistics.get_chrom_file(chrom),
+                )
+                with genomic_resource.open_raw_file(
+                        chrom_stat_filepath, mode="r") as infile:
+                    chrom_statistics[chrom] = ChromosomeStatistic.deserialize(
+                        infile.read()
+                    )
+        except FileNotFoundError:
+            logger.exception(
+                "Couldn't load statistics of %s", genomic_resource.resource_id
+            )
+            return ReferenceGenomeStatistics(genomic_resource, None, {})
+        return ReferenceGenomeStatistics(
+            genomic_resource,
+            global_statistic,
+            chrom_statistics
+        )
+
+
+class ChromosomeStatistic(Statistic):
+    """Class for individual chromosome statistics."""
+
+    def __init__(self, chromosome, length=0, nucleotide_counts=None):
+        super().__init__(chromosome, None)
+        if nucleotide_counts is None:
+            self.nucleotide_counts = {
+                "A": 0,
+                "G": 0,
+                "C": 0,
+                "T": 0,
+                "N": 0
+            }
+        else:
+            assert set(
+                nucleotide_counts.keys
+            ) == set(
+                ["A", "G", "C", "T", "N"]
+            )
+            self.nucleotide_counts = nucleotide_counts
+
+        self.nucleotide_distribution = {}
+        self.length = length
+        self.test = ""
+
+    def add_value(self, value):
+        self.length += 1
+        assert value in self.nucleotide_counts, value
```

Comment:
It happens that this assert fails on the real reference genomes. It is better to produce a warning message and skip the value:
```
        if value not in self.nucleotide_counts:
            logger.warning(
                "unexpected nucleotide <%s> in chromosome <%s>",
                value, self.statistic_id)
            return
```